src/cli/cli_link_metrics.cpp

```diff
@@ -0,0 +1,532 @@
+
+namespace ot {
+namespace Cli {
+
+LinkMetrics::LinkMetrics(otInstance *aInstance, OutputImplementer &aOutputImplementer)
+    : Output(aInstance, aOutputImplementer)
+    , mLinkMetricsQueryInProgress(false)
+{
+}
+
+template <> otError LinkMetrics::Process<Cmd("query")>(Arg aArgs[])
+{
+    otError       error = OT_ERROR_NONE;
+    otIp6Address  address;
+    bool          isSingle;
+    bool          blocking;
+    uint8_t       seriesId;
+    otLinkMetrics linkMetrics;
+
+    SuccessOrExit(error = aArgs[0].ParseAsIp6Address(address));
+
+    /**
+     * @cli linkmetrics query single
+     * @code
+     * linkmetrics query fe80:0:0:0:3092:f334:1455:1ad2 single qmr
+     * Done
+     * > Received Link Metrics Report from: fe80:0:0:0:3092:f334:1455:1ad2
+     * - LQI: 76 (Exponential Moving Average)
+     * - Margin: 82 (dB) (Exponential Moving Average)
+     * - RSSI: -18 (dBm) (Exponential Moving Average)
+     * @endcode
+     * @cparam linkmetrics query @ca{peer-ipaddr} single [@ca{pqmr}]
+     * - `peer-ipaddr`: Peer address.
+     * - [`p`, `q`, `m`, and `r`] map to #otLinkMetrics.
+     *   - `p`: Layer 2 Number of PDUs received.
+     *   - `q`: Layer 2 LQI.
+     *   - `m`: Link Margin.
+     *   - `r`: RSSI.
+     * @par
+     * Perform a Link Metrics query (Single Probe).
+     * @sa otLinkMetricsQuery
+     */
+    if (aArgs[1] == "single")
+    {
+        isSingle = true;
+        SuccessOrExit(error = ParseLinkMetricsFlags(linkMetrics, aArgs[2]));
+    }
+    /**
+     * @cli linkmetrics query forward
+     * @code
+     * linkmetrics query fe80:0:0:0:3092:f334:1455:1ad2 forward 1
+     * Done
+     * > Received Link Metrics Report from: fe80:0:0:0:3092:f334:1455:1ad2
+     * - PDU Counter: 2 (Count/Summation)
+     * - LQI: 76 (Exponential Moving Average)
+     * - Margin: 82 (dB) (Exponential Moving Average)
+     * - RSSI: -18 (dBm) (Exponential Moving Average)
+     * @endcode
+     * @cparam linkmetrics query @ca{peer-ipaddr} forward @ca{series-id}
+     * - `peer-ipaddr`: Peer address.
+     * - `series-id`: The Series ID.
+     * @par
+     * Perform a Link Metrics query (Forward Tracking Series).
+     * @sa otLinkMetricsQuery
+     */
+    else if (aArgs[1] == "forward")
+    {
+        isSingle = false;
+        SuccessOrExit(error = aArgs[2].ParseAsUint8(seriesId));
+    }
+    else
+    {
+        ExitNow(error = OT_ERROR_INVALID_ARGS);
+    }
+
+    blocking = (aArgs[3] == "block");
+
+    SuccessOrExit(error = otLinkMetricsQuery(GetInstancePtr(), &address, isSingle ? 0 : seriesId,
+                                             isSingle ? &linkMetrics : nullptr, HandleLinkMetricsReport, this));
+
+    if (blocking)
+    {
+        mLinkMetricsQueryInProgress = true;
+        error                       = OT_ERROR_PENDING;
+    }
+exit:
+    return error;
+}
+
+template <> otError LinkMetrics::Process<Cmd("mgmt")>(Arg aArgs[]) { return ProcessLinkMetricsMgmt(aArgs); }
```

Comment:
We can consider removing `ProcessLinkMetricsMgmt()` and have its implementation directly here?